Userland/Utilities/cksum.cpp

```diff
@@ -47,12 +47,22 @@ int main(int argc, char** argv)
             fail = true;
             continue;
         }
-        auto file_buffer = file->read_all();
-        auto bytes = file_buffer.bytes().size();
+        struct stat st;
+        if (fstat(file->fd(), &st) < 0) {
+            perror("fstat");
+            fail = true;
+            continue;
+        }
         if (algorithm == "crc32") {
-            outln("{} {} {}", Crypto::Checksum::CRC32 { file_buffer.bytes() }.digest(), bytes, path);
+            Crypto::Checksum::CRC32 crc32;
+            while (!file->eof() && !file->has_error())
+                crc32.update(file->read(PAGE_SIZE));
```

Comment:
I know it didn't have it before, but would some error handling make sense here?

Reply:
Not sure, I just copy pasted it from `checksum` program.

Reply:
Well, if the read fails for some reason (e.g. an I/O error) we'd just silently output a hash that was calculated for parts of the file. How about we output an error message in that case? :)

Reply:
Something like this? Also changed `perror()` to `warnln() ` to match other errors.

Reply:
That's so much better. :partying_face: